torch_xla/csrc/tensor_util.cpp

```diff
@@ -926,8 +926,8 @@ std::vector<torch::lazy::BackendDataPtr> CreateTensorsData(
       // across global devices for multi-host training.
       std::vector<std::string> local_devices =
           xla::ComputationClient::Get()->GetLocalDevices();
-      std::vector<at::Tensor> shards =
-          ShardingUtil::ShardTensor(tensors[i], sharding, local_devices);
+      std::vector<at::Tensor> shards = ShardingUtil::ShardTensor(
+          tensors[i], sharding, local_devices, /*padded=*/true);
```

Comment:
can you leave a comment explain why a padding is needed here?

Reply:
Done